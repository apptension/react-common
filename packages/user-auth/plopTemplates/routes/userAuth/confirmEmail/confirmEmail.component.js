import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Container,
} from '../../../components/forms/form.styles';
import messages from './confirmEmail.messages';

export class ConfirmEmail extends PureComponent {
  static propTypes = {
    match: PropTypes.object.isRequired,
    confirmEmail: PropTypes.func.isRequired,
  };

  componentWillMount() {
    const { match: { params: { uid, token } }, confirmEmail } = this.props;
    confirmEmail(uid, token);
  }

  render() {
    return (
      <Container>
        <FormattedMessage {...messages.confirming} />
      </Container>
    );
  }
}
