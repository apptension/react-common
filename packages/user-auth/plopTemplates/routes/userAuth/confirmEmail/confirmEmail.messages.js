import { defineMessages } from 'react-intl';

export default defineMessages({
  confirming: {
    id: 'register.confirming',
    defaultMessage: 'Confirming Email, please wait',
  },
});
