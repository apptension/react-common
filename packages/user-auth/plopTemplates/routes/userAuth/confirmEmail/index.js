export { default as ConfirmEmail } from './confirmEmail.container';
