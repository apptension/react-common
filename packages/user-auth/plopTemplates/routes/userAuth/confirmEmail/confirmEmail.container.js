import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';

import { UserAuthActions } from '../../../modules/userAuth';
import { ConfirmEmail } from './confirmEmail.component';

const mapStateToProps = createStructuredSelector({});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  confirmEmail: UserAuthActions.confirmEmail,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmEmail);
