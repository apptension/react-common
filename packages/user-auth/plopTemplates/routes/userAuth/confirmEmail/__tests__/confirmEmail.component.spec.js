import React from 'react';
import { shallow } from 'enzyme';
import { expect } from 'chai';
import { identity } from 'ramda';
import { spy } from 'sinon';

import { ConfirmEmail } from '../confirmEmail.component';

describe('<ConfirmEmail />', () => {
  const defaultProps = {
    match: {
      params: {
        uid: 'fakeuid',
        token: 'faketoken',
      },
    },
    confirmEmail: identity,
  };

  const component = (props) => shallow(
    <ConfirmEmail {...defaultProps} {...props} />
  );

  it('should render itself', () => {
    const wrapper = component();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should call confirmEmail on componentWillMount', () => {
    const confirmEmail = spy();
    component({ confirmEmail });
    expect(confirmEmail).to.have.been.calledOnce;
  });
});
