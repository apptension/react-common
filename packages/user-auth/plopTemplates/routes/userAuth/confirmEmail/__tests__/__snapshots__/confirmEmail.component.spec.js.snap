// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmEmail /> should render itself 1`] = `
<styled.div>
  <FormattedMessage
    defaultMessage="Confirming Email, please wait"
    id="register.confirming"
    values={Object {}}
  />
</styled.div>
`;
